ls1intum/Artemis
# `Programming exercises`: Allow administrators to configure the used docker build images for programming languages

### Checklist
#### General


#### Server
~- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).~
(Not needed. Reuise of old tests)
#### Changes affecting Programming Exercises
~- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).~
~- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).~
I've tested it on our local machines.

### Motivation and Context


Since different universities may use slightly different docker images for the builds of their programming exercises, this PR provides a simple mechanism to define the build images of the programming exercises via configuration files.

### Description

I've created a Configuration Property that contains a map for the build images. This map can be set in the known application-artemis.yml to the images you like. Therefore, the mapping is not defined at compile time but at start time.

Since the configuration files are loaded hierarchically, you should be able to override single language configuration by using the default values bundled in the configuration file in the artemis.war (that's something that should be tested)

### Steps for Testing

Prerequisites:
- 1 Admin

1. Change the Docker Image of some arbitrary language using a local configuration file.
2. Create a programming task with this language
3. Check the build configuration (Jenkins or Bamboo) (your defined docker image should be set)

### Review Progress





#### Code Review
#### Manual Tests
